test/unit_test.c

```diff
@@ -498,26 +498,55 @@ static void eh1(struct mg_connection *c, int ev, void *ev_data, void *fn_data) {
 }
 
 struct fetch_data {
-  char *buf;
-  int code, closed;
+  struct mg_iobuf *buf;
+  int code, closed, done;
 };
 
 static void fcb(struct mg_connection *c, int ev, void *ev_data, void *fn_data) {
   struct fetch_data *fd = (struct fetch_data *) fn_data;
   if (ev == MG_EV_HTTP_MSG) {
     struct mg_http_message *hm = (struct mg_http_message *) ev_data;
-    snprintf(fd->buf, FETCH_BUF_SIZE, "%.*s", (int) hm->message.len,
-             hm->message.ptr);
+    if (fd->buf->len == 0) {
+      size_t n;
+      n = fd->buf->size - fd->buf->len;
+      if (hm->message.len <= n)
+        n = hm->message.len;
+      mg_iobuf_add(fd->buf, fd->buf->len, hm->message.ptr, n, MG_IO_SIZE);
+    }
     fd->code = atoi(hm->uri.ptr);
     fd->closed = 1;
+    fd->done = 1;
     c->is_closing = 1;
+    fd->buf->buf[fd->buf->len] = '\0';
     (void) c;
+  } else if (ev == MG_EV_HTTP_CHUNK) {
```

Comment:
@cpq I modified the fetch callback to support `MG_EV_HTTP_CHUNK` so that it's able to process downloads larger than `MG_MAX_RECV_BUF_SIZE`, this fixes the failure in #1543.